internal/config/config.go

```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/harvester/terraform-provider-harvester/pkg/client"
+)
+
+type Config struct {
+	Bootstrap   bool
+	APIURL      string
```

Comment:
Personally, I would make this part of the bootstrap resource.
Thereby the `Config` structure can not contain contradicting information between this field and the API url that is encoded in the kubeconfig in the field below.
Since the API url is ignored in non-bootstrap mode, it create a confusing situation.

Reply:
Moved it to bootstrap resource.